### ISSUE TITLE
[CT-1997] Add a view of all public models with the same group

### ISSUE THREAD

**Issue by MichelleArk** [dbt-labs/dbt-docs#379]

**Comment by emmyoop** [comment]
Is this the type of thing we're thinking?  This shows 2 separate groups.  Only the models no marked as private are showing.  It is just a list of the models, ignoring where the files live.

![Screen Shot 2023-02-27 at 4 40 47 PM](https://user-images.githubusercontent.com/7070049/221702418-389b2785-f301-4713-8063-579322b4cf9c.png)

**Comment by MichelleArk** [comment]
Yes! This is exactly what I was thinking. Now that we've got protected in addition to public and private access, I wonder if it'll be confusing to call this view 'Public Models`. Especially if we're only excluding `private` models in this view. 

Users looking at the docs from other projects will be interested in only public models, whereas users within the project will be interested in _both_ public and protected models. Perhaps we continue to call the view 'Public Models', include both public and protected models, but distinguish protected models in the model name? Something like:

```
Public Models: 
nested_group/
  top-model-public
  top-model (protected)
some_group/
  even_id_table
```

**Comment by emmyoop** [comment]
Updated icon and added a `(protected)` string at the end of models.

![Screen Shot 2023-02-28 at 9 57 36 AM](https://user-images.githubusercontent.com/7070049/221971065-b2275e69-4737-4fcb-b5e2-063c6f1064aa.png)
